cpp/common/src/codingstandards/cpp/deadcode/UnusedVariables.qll

```diff
@@ -56,6 +56,16 @@ class PotentiallyUnusedLocalVariable extends LocalVariable {
   }
 }
 
+/**
+ * A `LocalVariable` which is a candidate for being unused, and not defined in a macro.
+ */
+class PotentiallyUnusedLocalVariable extends BasePotentiallyUnusedLocalVariable {
+  PotentiallyUnusedLocalVariable() {
+    // Ignore variables declared in macro expansions
+    not exists(DeclStmt ds | ds.getADeclaration() = this and ds.isInMacroExpansion())
```

Comment:
The existing queries simply ignore variables declared in macro expansions - we should add an issue to review backporting the macro handling capabilities from this PR.

Reply:
filed: https://github.com/github/codeql-coding-standards/issues/838